pkg/stub/handler.go

```diff
@@ -349,16 +350,27 @@ func (h *Handler) tbrInaccessible() bool {
 		}
 	}
 	err = wait.PollImmediate(5*time.Second, 3*time.Minute, func() (bool, error) {
+		// we have seen cases in the field with disconnected cluster where the default connection timeout can be
+		// very long (15 minutes in one case); so we do an initial non-tls connection were we can specify a quicker
+		// timeout to filter out that scenario and default to tbr inaccessible / Removed in an expedient fashion
+		connWithTimeout, err := net.DialTimeout("tcp", "registry.redhat.io:443", 15*time.Second)
+		if err != nil {
+			logrus.Infof("test connection with timeout failed with %s", err.Error())
+			return false, err
+		}
+		defer connWithTimeout.Close()
+		// still do the tls form of connect (which does not have the handy timeout form of dial) to confirm
+		// ssl handshake is OK
 		tlsConf := &tls.Config{}
 		conn, err := tls.Dial("tcp", "registry.redhat.io:443", tlsConf)
```

Comment:
I think you can use `tls.Client(connWithTimeout, tlsConf)` (maybe with minor tweaks: https://cs.opensource.google/go/go/+/refs/tags/go1.16.7:src/crypto/tls/tls.go;l=154-164;drc=refs%2Ftags%2Fgo1.16.7)

Reply:
yep I'll add that - thanks @dmage

Reply:
after an iteration to addres ` either ServerName or InsecureSkipVerify must be specified in the tls.Config` I have it working

pushing update momentarily